addons/packages/metrics-server/0.5.1/bundle/config/values.yaml

```diff
@@ -12,7 +12,7 @@ daemonset:
   updateStrategy: null
 metricsServer:
   namespace: null
-  createNamespace: true
+  createNamespace: false
```

Comment:
Couple of scenarios we need to handle:
1. User giving createNamespace: true and namespace: kube-system - In this case lets not create namespace
2. Upgrade: User upgrades the package but has older data values i.e. User in their data values has createNamespace: true. So even though package defaults are changed the user has createNamespace: true and hence during delete it will attempt to delete the namespace. Lets also add `https://carvel.dev/kapp/docs/v0.46.0/apply/#kappk14siodelete-strategy` delete-strategy:orphan in the namespace to prevent deletion of namespace

Reply:
namespace creation is handled by kapp-controller. I don't tink we can do much here to prevent the creation or adoption of the namespace when the user intentionally sets createNamespace: true. 
However adding delete-strategy:orphan to the namespace will mitigate this as well.

Reply:
It looks like this behavior has been fixed in https://github.com/vmware-tanzu/carvel-kapp/pull/468


see discussion here: https://kubernetes.slack.com/archives/C02GY94A8KT/p1645018465370589
There is a request to add a feature to kapp to prevent a package from taking ownership: https://github.com/vmware-tanzu/carvel-kapp-controller/issues/523

and the [fix](https://github.com/vmware-tanzu/carvel-kapp/pull/468) is referenced there. 
looks like fix should be available as of  kapp v0.47.0

Reply:
for deployments that use older kapp-controller the best approach will be to add https://carvel.dev/kapp/docs/v0.46.0/apply/#kappk14siodelete-strategy delete-strategy:orphan, and modify the values.yaml as suggested in this pr

Reply:
1. Lets add https://carvel.dev/kapp/docs/v0.46.0/apply/#kappk14siodelete-strategy delete-strategy:orphan for older kapp-controller deployments
2. Can we add a check here  https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/metrics-server/0.5.1/bundle/config/overlays/overlay-namespace.yaml#L12 
`#@ if data.values.metricsServer.createNamespace and data.values.namespace != "kube-system":` 
To prevent even creating namespace if kube-system is given ?

Reply:
sounds like a good plan